environments/base/manifests/loki-distributor-grpc-service.yaml

```diff
@@ -0,0 +1,21 @@
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app.kubernetes.io/instance: observatorium-xyz
+    app.kubernetes.io/name: loki
+    app.kubernetes.io/part-of: observatorium
+    app.kubernetes.io/version: master-815c475
+  name: observatorium-xyz-loki-distributor-grpc
```

Comment:
why don't we combine the various services that select one single set of pods (e.g. `-grpc` and `-http`) into the same service and instead specify multiple port combinations? I think this would reduce some mental overhead

i.e. we have a loki-distributor service, a loki-frontend service, and a loki-querier service rather than 2-3 for each of those components

Reply:
The reason why we have 2 services is reflected in GRPC being responsible for load-balancing itself. Thus for GRPC communication we need a headless service. However, for incoming HTTP traffic we keep the k8s load-balancing via ClusterIP. The gossip service is a cross-component service that constraints communication only between distributors, ingesters and queriers based on an extended label selector set. IMHO it is not possible to unify all three demand in one service, e.g. enabling ClusterIP for the HTTP port while headless for the GRPC. For more info please check above the discussion with @cyriltovena:

Reply:
ack sounds good :+1: